DXMainClient/DXGUI/Multiplayer/CnCNet/ChoiceNotificationBox.cs

```diff
@@ -0,0 +1,189 @@
+            lblSender.FontIndex = 1;
+            lblSender.ClientRectangle = new Rectangle(gameIconPanel.Right + 3, lblHeader.Bottom + 6, 0, 0);
+            lblSender.Text = "fonger";
+            AddChild(lblSender);
+
+            lblChoiceText = new XNALabel(WindowManager);
+            lblChoiceText.Name = "lblChoiceText";
+            lblChoiceText.FontIndex = 1;
+            lblChoiceText.ClientRectangle = new Rectangle(12, lblSender.Bottom + 6, 0, 0);
+            lblChoiceText.Text = "What do you want to do?";
+            AddChild(lblChoiceText);
+
+            affirmativeButton = new XNAButton(WindowManager);
+            affirmativeButton.FontIndex = 1;
+            affirmativeButton.ClientRectangle = new Rectangle(ClientRectangle.Left + 8, lblChoiceText.Bottom + 6, 75, 23);
+            affirmativeButton.IdleTexture = AssetLoader.LoadTexture("75pxbtn.png");
+            affirmativeButton.HoverTexture = AssetLoader.LoadTexture("75pxbtn_c.png");
+            affirmativeButton.HoverSoundEffect = new EnhancedSoundEffect("button.wav");
+            affirmativeButton.Name = "affirmativeButton";
+            affirmativeButton.Text = "Yes";
+            affirmativeButton.LeftClick += AffirmativeButton_LeftClick;
+            AddChild(affirmativeButton);
+
+            negativeButton = new XNAButton(WindowManager);
+            negativeButton.FontIndex = 1;
+            negativeButton.ClientRectangle = new Rectangle(ClientRectangle.Width - (75 + 8), lblChoiceText.Bottom + 6, 75, 23);
+            negativeButton.IdleTexture = AssetLoader.LoadTexture("75pxbtn.png");
+            negativeButton.HoverTexture = AssetLoader.LoadTexture("75pxbtn_c.png");
+            negativeButton.HoverSoundEffect = new EnhancedSoundEffect("button.wav");
+            negativeButton.Name = "negativeButton";
+            negativeButton.Text = "No";
+            negativeButton.LeftClick += NegativeButton_LeftClick;
+            AddChild(negativeButton);
+
+            base.Initialize();
+        }
+
+        public void Show(string headerText, Texture2D gameIcon, string sender, string choiceText, string affirmativeText, string negativeText, int timeout)
+        {
+            Visible = true;
+            Enabled = true;
+
+            lblHeader.Text = headerText;
+            gameIconPanel.BackgroundTexture = gameIcon;
+            lblSender.Text = sender;
+            lblChoiceText.Text = choiceText;
+            affirmativeButton.Text = affirmativeText;
+            negativeButton.Text = negativeText;
+
+            // use the same clipping logic as the PM notification
+            if (lblChoiceText.Width > Width)
+            {
+                while (lblChoiceText.Width > Width)
+                {
+                    lblChoiceText.Text = lblChoiceText.Text.Remove(lblChoiceText.Text.Length - 1);
+                }
+            }
+
+            downTime = TimeSpan.Zero;
+            isDown = true;
+
+            downTimeWaitTime = TimeSpan.FromSeconds(timeout);
```

Comment:
Do those choice notif-s really need to time out in all cases? My suggestion is to make that optional (maybe set `timeout = 0` as a default argument value and don't check for wait time, so the notif. stays indefinitely).